v3/routes/exercises.rb

```diff
@@ -18,6 +20,14 @@ class Exercises < Core
         status 404 unless problem.exists?
         pg :"exercise/exists", locals: { problem: problem }
       end
+
+      get '/tracks/:id/exercises/:slug' do |id, slug|
+        problem = config.find(id).find(slug)
+        file = Tempfile.new('data.zip')
```

Comment:
I think `Tempfile` is the right approach. I'm actually not sure we're allowed to access the file system on heroku, so we'll need to do a test to verify that this will work there.